src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -51,6 +52,17 @@ export const Configurator = ({
     return languages[0];
   });
 
+  const shouldDisplayTimeFieldSelector = (ds: Dataset, lang: string, fields: DatasetField[]) => {
+    return (
+      !(
+        languageService.getLanguage(lang)?.disableDatePicker ||
+        ds.type === DEFAULT_DATA.SET_TYPES.INDEX_PATTERN
+      ) &&
+      fields &&
+      fields.length > 0
+    );
+  };
```

Comment:
```suggestion
  const submitDisabled = useMemo (() => {
    return !(
      timeFieldName === undefined &&
      !(
        languageService.getLanguage(lang)?.disableDatePicker ||
        dataset.type === DEFAULT_DATA.SET_TYPES.INDEX_PATTERN
      ) &&
      timeFields &&
      timeFields.length > 0
    );
  }, [dataset, language, timeFieldName, timeFields, languageService]);
```
nit: Also if you use the logic i have suggested before, this name probably wont be appropriate anymore. Then its only usecase is to disable the select button on the bottom. And for that, id just keep the logic colocated here.

Reply:
Adding this logic in 51e4490